app/models.py

```diff
@@ -388,6 +391,17 @@ def serialize(self, data=None):
             "countersignedAt": countersigned_at,
         }, **(data or {}))
 
+        if self.agreement_details and self.agreement_details.get('uploaderUserId'):
+            user = User.query.filter(
+                User.id == self.agreement_details.get('uploaderUserId')
+            ).first()
+
+            if user:
```

Comment:
This isn't just to do with what CCS will see though - if our app is sending a user ID that doesn't exist then something is fundamentally broken and we shouldn't proceed to save something wrong.
The case of serializing an audit event is not the same thing at all - that's deciding about what data to *show* that has already been stored in the database, and it's possible that some audit events might not have an associated user (peraps they have been deleted several years down the line).
But in this case it's about what info to save into our DB, and a user doing an action right now should definitely exist.